test/Tests/src/Data/Statistics_Spec.enso

```diff
@@ -179,6 +221,12 @@ spec =
             no_ord_set.compute Statistic.Minimum . should_fail_with Incomparable_Values
             no_ord_set.compute Statistic.Maximum . should_fail_with Incomparable_Values
 
+        Test.specify "should fail with the underlying error on broken comparator" <|
+            bad_comparator_set = [Bad_Comparator_Type.Value 10, Bad_Comparator_Type.Value 2, Nothing, Bad_Comparator_Type.Value 9]
+            bad_comparator_set.compute . should_equal 3
+            bad_comparator_set.compute Statistic.Minimum . should_fail_with Illegal_State
+            bad_comparator_set.compute Statistic.Maximum . should_fail_with Illegal_State
```

Comment:
Let's add a test for `running` also. My first look I was thinking it will not be propagated, now I see `throw_wrapped_if_error` so it looks like it will work, but I imagine it's good to test it.

```suggestion
            bad_comparator_set.compute Statistic.Maximum . should_fail_with Illegal_State
            bad_comparator_set.running Statistic.Maximum . should_fail_with Illegal_State
```